cranelift/codegen/src/ir/libcall.rs

```diff
@@ -140,13 +140,28 @@ impl LibCall {
                 sig.params.push(AbiParam::new(ty));
                 sig.returns.push(AbiParam::new(ty));
             }
-            LibCall::Probestack
-            | LibCall::Memcpy
-            | LibCall::Memset
-            | LibCall::Memmove
-            | LibCall::Memcmp
-            | LibCall::ElfTlsGetAddr
-            | LibCall::ElfTlsGetOffset => unimplemented!(),
+            LibCall::Memcpy | LibCall::Memmove => {
+                sig.params.push(AbiParam::new(pointer_type));
+                sig.params.push(AbiParam::new(pointer_type));
+                sig.params.push(AbiParam::new(pointer_type));
+                sig.returns.push(AbiParam::new(pointer_type));
+            }
+            LibCall::Memset => {
+                sig.params.push(AbiParam::new(pointer_type));
+                sig.params.push(AbiParam::new(I32));
+                sig.params.push(AbiParam::new(pointer_type));
+                sig.returns.push(AbiParam::new(pointer_type));
+            }
+            LibCall::Memcmp => {
+                sig.params.push(AbiParam::new(pointer_type));
+                sig.params.push(AbiParam::new(pointer_type));
+                sig.params.push(AbiParam::new(pointer_type));
+                sig.returns.push(AbiParam::new(I32))
+            }
```

Comment:
I had a really hard time figuring out why the length arguments also use `pointer_type` since they aren't pointers. You are correct but maybe we can make this a little more clear.

One way might be to rename that parameter; maybe `int_pointer_type`?

Another way is to just add a comment. Actually I'd kind of appreciate a comment on each libcall signature anyway. Maybe describing its API in either C or Rust syntax, as a reminder, so it's easy to compare that to the ABI we're transforming it to.

Reply:
Added the C function signatures in comments. I don't really like `int_pointer_type` as name as I think it can be interpreted as either a type that is used for `int*` pointers or for the `int` type and pointers, but it is used for `size_t` and pointers.